src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1193,6 +1197,88 @@ protected override void HandlePauseRequestAsync(IRequestResponder<PauseArguments
             m_program.CauseBreak();
             responder.SetResponse(new PauseResponse());
         }
+        
+        protected override void HandleGotoRequestAsync(IRequestResponder<GotoArguments> responder)
+        {
+            var response = new GotoResponse();
+            if (!m_isStopped)
+            {
+                responder.SetResponse(response);
+                return;
+            }
+
+            var gotoTarget = m_gotoCodeContexts[responder.Arguments.TargetId];
+            IDebugThread2 thread = null;
+            lock (m_threads)
+            {
+                if (!m_threads.TryGetValue(responder.Arguments.ThreadId, out thread))
+                    throw new AD7Exception("Could not find thread!");
+            }
+            BeforeContinue();
+            var builder = new ErrorBuilder(() => AD7Resources.Error_UnableToSetNextStatement);
+            try
+            {
+                builder.CheckHR(thread.SetNextStatement(null, gotoTarget));
+            }
+            catch (AD7Exception)
+            {
+                m_isStopped = true;
+                throw;
```

Comment:
throwing here will stop the OpenDebugAD7 process, you will want to use `responder.SetError(new ProtocolException(e.Message));` so the error is sent to the client.